bundle/regal/rules/style/external-reference/external_reference.rego

```diff
@@ -33,6 +33,15 @@ report contains violation if {
 	violation := result.fail(rego.metadata.chain(), result.location(value))
 }
 
+_args_vars(fn) := {name |
+	some arg in fn.head.args
+	some name in _named_vars(arg)
+}
+
+_named_vars(arg) := {arg.value} if arg.type == "var"
+
+_named_vars(arg) := {var.value | some var in ast.find_term_vars(arg)} if arg.type in {"array", "object", "set"}
```

Comment:
Huh it never occurred to me to use a _set_ as a function argument. 💡 

```rego
foo({1, 2, 3}) if true

res := foo({3, 2, 1})
```
🔀 `res = true`. Nice.

Reply:
hehe, yeah... variable bindings are odd in that context though

```rego
package play

import rego.v1

f({a, b}) := a + b

x := f({1, 2})
```

that just comes back as undefined.. pattern matching is obviously hard to do for an allegedly unordered collection... but should it be an error in that case? 🤔

Reply:
Hmm yeah... 
```rego
package play
import rego.v1

f({1, b}) := 1 + b

x := f({1, 2})
```
Too bad, this would have been cool. Works for arrays,
```rego
f([1, b]) := 1 + b

x := f([1, 2])
```

....but I think there's an issue for set unification already...

Reply:
No issue, but it's [documented:](https://www.openpolicyagent.org/docs/latest/policy-language/#composite-values)
<img width="1002" alt="image" src="https://github.com/user-attachments/assets/1aef1c01-ba25-4fb7-be3e-c48b7158f6f1">

Reply:
That's helpful, thanks! Although I think what I would have expected here was precisely that OPA would complain 🙂 I guess we're not the only ones not having thought of this too much before, hehe

Reply:
That particular screenshot is misleading in a few ways: It's talking about _unification_ but uses "==".  :/